### PR TITLE
Add submission graphs to discussion details in teacher

### PR DESCRIPTION
**Pull request by vanwagonet**
refs: MBL-14386
affects: Teacher
release note: none

### CODE REVIEW

**Comment by vanwagonet** [comment]
Test Plan:
In teacher, view a discussion assignment, verify the graphs are shown and work like they do in RN
View a non-assignment discussion, verify the submission section is not shown
In Student verify the submission section is not shown

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/mkqu97m) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fmkqu97m)
teacher: [Link](https://rebrand.ly/gdftipp) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fgdftipp)
parent: [Link](https://rebrand.ly/h7qfvnh) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fh7qfvnh)

File: Core/Core/Discussions/DiscussionDetailsViewController.swift [hunk 1]
```diff
@@ -215,18 +218,30 @@ public class DiscussionDetailsViewController: UIViewController, ColoredNavViewPr
...
-        }
+        let isPublished = topic.first?.published == true
+        publishedIcon.image = .icon(isPublished ? .publish : .no, .solid)
+        publishedIcon.tintColor = .named(isPublished ? .textSuccess : .textDark)
+        publishedLabel.text = isPublished
+            ? NSLocalizedString("Published", bundle: .core, comment: "")
+            : NSLocalizedString("Unpublished", bundle: .core, comment: "")
+        publishedLabel.textColor = .named(isPublished ? .textSuccess : .textDark)
         publishedView.isHidden = env.app != .teacher || isAnnouncement || showRepliesToEntryID != nil
+
```
Comment (inst-danger) [inline Core/Core/Discussions/DiscussionDetailsViewController.swift:229]:
```suggestion

```

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Teacher

[MBL-14386](https://instructure.atlassian.net/browse/MBL-14386)

<details><summary>:x: Swift lint</summary>


    ⚠️  /Users/vagrant/git/Core/Core/Discussions/DiscussionDetailsViewController.swift:229:1: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    Linting 'PlannablesRequestable.swift' (365/758)
    Linting 'PlannerNoteDetailViewController.swift' (366/758)
    ⚠️  /Users/vagrant/git/Core/Core/Submissions/SubmissionBreakdownViewController.swift:109:5: Large Tuple Violation: Tuples should have at most 2 members. (large_tuple)
    /Users/vagrant/git/Core/Core/Submissions/SubmissionBreakdownViewController.swift:110:5: warning: Large Tuple Violation: Tuples should have at most 2 members. (large_tuple)
    Linting 'FileUploadContext.swift' (499/758)
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by vanwagonet** [comment]
Found an issue with group discussions, will fix that first.

File: Core/Core/Discussions/DiscussionDetailsViewController.swift [hunk 2]
```diff
@@ -232,9 +232,9 @@ public class DiscussionDetailsViewController: UIViewController, ColoredNavViewPr
                 dueSection.isHidden = false
             }
 
-            if topic.first?.groupTopicChildren == nil, submissionsSection.isHidden {
+            if topic.first?.groupTopicChildren == nil, submissionsSection.isHidden, let courseID = courseID {
                 let controller = SubmissionBreakdownViewController.create(
-                    context: context,
+                    courseID: courseID,
```
Comment (vanwagonet) [inline Core/Core/Discussions/DiscussionDetailsViewController.swift:237]: made this always require a courseID to make sure we don't try to use the group context.

File: Core/Core/Discussions/DiscussionHTML.swift [hunk 3]
```diff
@@ -234,7 +234,7 @@ enum DiscussionHTML {
             entries.length > 0 && h('h2', { class: \(s(.heading)) },
                 \(s(NSLocalizedString("Replies", bundle: .core, comment: "")))
             ),
-            entries.map(entry => h(Discussion.Entry, { topic, entry, depth: 0, maxDepth, canLike }))
+            entries.map(entry => h(Discussion.Entry, { key: entry.id, topic, entry, depth: 0, maxDepth, canLike }))
```
Comment (vanwagonet) [inline Core/Core/Discussions/DiscussionHTML.swift:237]: keys should help with update performance as things change.

File: Core/Core/Submissions/SubmissionBreakdownViewController.swift [hunk 4]
```diff
@@ -95,8 +95,8 @@ class SubmissionBreakdownViewController: UIViewController {
         summary.refresh()
     }
 
-    override func viewDidAppear(_ animated: Bool) {
-        super.viewDidAppear(animated)
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
```
Comment (vanwagonet) [inline Core/Core/Submissions/SubmissionBreakdownViewController.swift:99]: starts the animation earlier to match RN.

File: Core/Core/Submissions/SubmissionBreakdownViewController.swift [hunk 5]
```diff
@@ -156,15 +156,15 @@ class SubmissionBreakdownViewController: UIViewController {
...
         gradedProgress.progress = CGFloat(graded / total)
 
         let ungraded = animateFrom.1 + (t * (animateTo.1 - animateFrom.1))
-        ungradedCountLabel.text = NumberFormatter.localizedString(from: NSNumber(value: ungraded), number: .none)
+        ungradedCountLabel.text = NumberFormatter.localizedString(from: NSNumber(value: floor(ungraded)), number: .none)
         ungradedProgress.progress = CGFloat(ungraded / total)
 
         let unsubmitted = animateFrom.2 + (t * (animateTo.2 - animateFrom.2))
-        unsubmittedCountLabel.text = NumberFormatter.localizedString(from: NSNumber(value: unsubmitted), number: .none)
+        unsubmittedCountLabel.text = NumberFormatter.localizedString(from: NSNumber(value: floor(unsubmitted)), number: .none)
```
Comment (vanwagonet) [inline Core/Core/Submissions/SubmissionBreakdownViewController.swift:167]: floor makes the number never round up and get to the target number too early.

File: Core/Core/Views/CoreWebView.swift [hunk 6]
```diff
@@ -249,6 +249,7 @@ open class CoreWebView: WKWebView {
                 checkSize()
                 document.addEventListener('load', checkSize, true)
             })
+            window.addEventListener('error', checkSize, true)
```
Comment (vanwagonet) [inline Core/Core/Views/CoreWebView.swift:252]: if there is a broken image in the discussion the sizing wasn't quite right without this.

File: Student/Student/Student-Bridging-Header.h [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
```
Comment (vanwagonet) [inline Student/Student/Student-Bridging-Header.h:17]: this is necessary to work around the debugging issues with PSPDFKit.

**Pull request merged**